cibyl/sources/zuul/source.py

```diff
@@ -64,12 +64,8 @@ def get_model_for(job):
                 )
 
                 if fetch_builds:
-                    builds = self._get_builds_for(job, **kwargs)
-
-                    model.builds = {
-                        build['uuid']: Build(build['uuid'], build['result'])
-                        for build in builds
-                    }
+                    for build in self._get_builds_for(job, **kwargs):
+                        model.add_build(Build(build['uuid'], build['result']))
```

Comment:
The old solution was broken, this one uses the model interface so as to not break anything.